mlflow/utils/annotations.py

```diff
@@ -21,15 +22,23 @@ def deprecated(alternative=None, since=None):
     :param func: A function to mark
     :returns Decorated function.
     """
-    def deprecated_func(func):
-        since_str = " since %s" % since if since else ""
-        notice = ".. Warning:: Deprecated%s: This method will be removed in " % since_str + \
-                 "a near future release."
-        if alternative is not None and alternative.strip():
-            notice += " Use ``%s`` instead." % alternative
-        func.__doc__ = notice + "\n" + func.__doc__
-        return func
-    return deprecated_func
+
+    def deprecated_decorator(func):
+        @wraps(func)
+        def deprecated_func(*args, **kwargs):
+            since_str = " since %s" % since if since else ""
+            notice = ".. Warning::" + \
+                     func.__name__ + " is deprecated%s: This method will be removed in "\
```

Comment:
Can we include the module name as well? Something like `".".join([func.__module__, func.__name__])`

Reply:
Ah, I was looking for something like that. Thanks!